CompilerSource/gcc_interface/gcc_backend.h

```diff
@@ -34,23 +34,14 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <sstream> // std::stringstream
+#include <fstream> // std::ifstream
 
 inline std::string fc(const char* fn) {
-  FILE* pt = fopen(fn, "rb");
-  if (pt == NULL)
-    return "";
-  else {
-    fseek(pt, 0, SEEK_END);
-    size_t sz = ftell(pt);
-    fseek(pt, 0, SEEK_SET);
-
-    char a[sz + 1];
-    sz = fread(a, 1, sz, pt);
-    fclose(pt);
-
-    a[sz] = 0;
-    return a;
-  }
+  std::ifstream t(fn);
+  std::stringstream buffer;
+  buffer << t.rdbuf();
```

Comment:
What is the behavior of this line when `!t.is_open()`?

Reply:
it returns an empty string. Do you want to add a warning for that case?

Reply:
Not a strong preference, but I can't find any documentation that defines that behavior. I can take a look at the standard to verify it, I guess. Printing the warning and returning the empty string guarantees that we're fine, though, so I certainly am not opposed to that.

Reply:
Does fc stand for fundie's cat?

Reply:
added warning for this case